grant/projects/offchain-subnames.md

```diff
@@ -0,0 +1,145 @@
+# Offchain Subnames on Darwinia Chain
+
+## Project Description
+
+This project aims to implement an offchain subnames solution for the 'darwinia.eth' Ethereum Name Service (ENS) domain but keeps the subname records on Darwinia Chain. 
+
+It leverages the EIP-3668 specification to enable efficient and cost-effective management of subname-address mappings off the Ethereum mainnet. By storing these mappings on Darwinia Chain, the project offers a scalable approach to handle a large number of subdomains without incurring high gas fees.
+
+The solution utilizes a customized OffchainResolver contract on Ethereum, which works with a gateway service connected to the Darwinia Chain. This setup allows for seamless resolution of subnames (e.g., 'foo.darwinia.eth') while maintaining the security and decentralization benefits of the ENS.
+
+This project addresses the need for efficient subdomain management within the Darwinia ecosystem, as outlined in the RFP (https://github.com/ringecosystem/collaboration/issues/61), and demonstrates the potential for cross-chain interoperability in domain name services.
+
+### This project will benefit the Darwinia ecosystem in the following ways:
+
+1. It will provide a more efficient and cost-effective way to manage subdomains on the Darwinia Chain, reducing the need for users to pay high gas fees on Ethereum.
+2. It will enhance the usability of the 'darwinia.eth' domain by allowing users to register, update, and resolve subnames with reduced costs and improved performance.
+3. It will demonstrate the potential for cross-chain interoperability in domain name services, opening up new possibilities for collaboration and innovation within the Darwinia ecosystem.
+
+## Team
+
+- Aki Wu([wuminzhe](https://github.com/wuminzhe))  
+  A full stack developer focused on the web3 space. He has been working on darwinia ecosystem for a long time.
+
+- Echo([Echo](https://github.com/hujw77))  
+  Contracts developer from Itering.
+
+- Bear([Bear](https://github.com/boundless-forest))
+  Project Manager from Itering. 
+  
+- Yuqi(yuqi.liu@itering.io)
+  UI Designer from Itering.
+
+## Team Code Repos
+
+- https://github.com/subnames/subnames-contracts
+  The repo for the subnames contracts based on [basenames](https://github.com/base-org/basenames).
+
+- https://github.com/subnames/offchain-resolver  
+  1. the offchain gateway.
+  2. the offchain resolve client.
+  3. based on ensdomains's [offchain-resolver](https://github.com/ensdomains/offchain-resolver).
+
+- https://github.com/wuminzhe/subnames-ui
+  Frontend User Interface.
+
+## Development Roadmap
+
+### Milestone 1
+
+Estimated Duration: 2 weeks
+
+Cost: 1000 USDT
+
+Address: 0x2Da8ccfe0dD2165B8d939eaBf4E3697C4Adb6FDd
+
+#### Tasks:
+
+1. Configure the contracts to support "ringdao.eth" domain for testing. 
```

Comment:
darwinia.eth

Reply:
milestone1 is a already running on koi network, I will deploy them to crab network after milestone2